torch/csrc/jit/passes/symbolic_shape_cache.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/passes/symbolic_shape_analysis.h>
+
+namespace torch {
+namespace jit {
+
+struct TORCH_API CanonicalizedSymbolicShape {
+  CanonicalizedSymbolicShape(
+      const c10::SymbolicShape& orig_shape,
+      std::unordered_map<int64_t, int64_t>& ss_map) {
+    init(orig_shape, ss_map);
+  }
+
+  CanonicalizedSymbolicShape(c10::SymbolicShape& orig_shape) {
+    std::unordered_map<int64_t, int64_t> new_ssmap;
+    init(orig_shape, new_ssmap);
+  }
+
+  size_t hash() const;
+
+  c10::SymbolicShape toSymbolicShape(
+      std::unordered_map<int64_t, int64_t> inverse_ss_map) const;
```

Comment:
this should be by reference `std::unordered_map<int64_t, int64_t>&`, and might as well take in const as well

Reply:
It can't be const, as you need to be able to insert into the map, both for two symbols in the same result tensor, and for the case where there can be symbols shared between two returned tensors. 
However, good call on the `&`, as that is a correctness issue (in the case of multiple returned tensors).

Reply:
do we have any tests that woudl catch that ?

Reply:
Added `TestShapeMultipleReturns` to catch this case.